elftools/elf/elffile.py

```diff
@@ -376,3 +398,40 @@ def _read_dwarf_section(self, section, relocate_dwarf_sections):
                 name=section.name,
                 global_offset=section['sh_offset'],
                 size=section['sh_size'])
+
+    def _decompress_dwarf_section(self, section):
+        """ Returns the uncompressed contents of the provided DWARF section.
+        """
+        # TODO: support other compression formats from readelf.c
+        assert section.size > 12, 'Unsupported compression format.'
+
+        section.stream.seek(0)
+        # According to readelf.c the content should contain "ZLIB"
+        # followed by the uncompressed section size - 8 bytes in
+        # big-endian order
+        compression_type = bytes2str(section.stream.read(4))
+        assert compression_type == 'ZLIB', \
+            'Invalid compression type: %r' % (compression_type)
+
+        uncompressed_size = 0
+        for _ in range(8):
```

Comment:
using the struct module here would be cleaner, I think. You probably need `struct.unpack` with the Q format